### DISCUSSION TITLE
Avatar uploads fail, but all other uploads are successful

### DISCUSSION THREAD

**Discussion by beefhooker** [outline/outline#5963] [Self Hosting]
Hi there, 

I've run into a strange error I was hoping someone could possibly point me in the right direction, please.

I've been able to successfully set up my S3 handled by an AWS bucket. Uploading images, file attachments, etc. inside documents work without issue.

However, trying to update/change the profile picture/avatar of any profile or the workspace returns an "Upload Failed" message.

The network tab in the developer console returns a 400 Bad Request error.

It's not a huge issue, but definitely curious and any help would be appreciated :)

**Comment by alkanna** [comment]
Can you see what the error is on the body associated to this 400 error ?
Can you see any error in the server logs as well ?

**Comment by AlexShku** [comment]
Hello! I have the same error! 
https://github.com/outline/outline/discussions/6241
How it can be fixed?

**Comment by nickcharlton** [comment]
I'm seeing the same thing — other uploads work fine, but avatar fails.

In my case, the 400 Bad Request is coming from line 67 in `app/utils/files.ts`: https://github.com/outline/outline/blob/62f3e6921fdf93c3f3003980a58fcfcca66a0ce1/app/utils/files.ts#L51-L67

---

Perhaps the error is different for everyone, but looking into the _Network_ tab of my browser, the error is showing as:

```
<Error>
  <Code>AccessControlListNotSupported</Code>
  <Message>The bucket does not allow ACLs</Message>
  <RequestId>VKT8SJZSS96RYXDC</RequestId>
  <HostId>ueHvJxImAQe5yvCOuS7UmEpyHlL/MYGDgWnI5LUyIosFj/RWcLdaPWDOfzJAy46XOp+UFlOHalw=</HostId>
</Error>
```

---

The trick which helped me resolve this was this discussion: https://github.com/outline/outline/discussions/6704

Or, in Terraform:

```
resource "aws_s3_bucket_ownership_controls" "example" {
  bucket = aws_s3_bucket.example.id

  rule {
    object_ownership = "BucketOwnerPreferred"
  }
}
```
